spec/interactors/process_hubee_notification_spec.rb

```diff
@@ -0,0 +1,190 @@
+    let(:status) { "SI_RECEIVED" }
+
+    before do
+      allow(HubEE::Api).to receive(:session).and_return(session)
+      allow(session).to receive(:event).and_return(event_payload)
+      allow(session).to receive(:update_event)
+      allow(session).to receive(:delete_notification)
+    end
+
+    # it "deletes the notification" do
+    #   expect(session).to receive(:delete_notification)
+    #   interactor
+    # end
+
+    context "when the notification is for FormulaireQF" do
+      context "when the notification is for an existing folder" do
+        context "when the event is fetch correctly" do
+          context "when the event is sent" do
+            context "when the event is a status update" do
+              let!(:shipment) { create(:shipment, hubee_case_id: "1234") }
+
+              it "update the hubee event" do
+                expect(session).to receive(:update_event)
+                interactor
+              end
+
+              context "when the event is processable" do
+                it "deletes the notification" do
+                  expect(session).to receive(:delete_notification)
+                  interactor
+                end
+
+                it "updates the shipment status" do
+                  expect { interactor }.to change { shipment.reload.hubee_status }.to("si_received")
+                end
+
+                context "when the event is a final status" do
+                  let(:status) { "DONE" }
+
+                  it "removes hubee ids" do
+                    expect {
+                      interactor
+                      shipment.reload
+                    }.to change { shipment.hubee_folder_id }.to(nil)
+                      .and change { shipment.hubee_case_id }.to(nil)
+                  end
+                end
+
+                context "when the event is not a final status" do
+                  it "does not remove hubee ids" do
+                    expect {
+                      interactor
+                      shipment.reload
+                    }.to not_change { shipment.hubee_folder_id }
+                      .and not_change { shipment.hubee_case_id }
+                  end
+                end
+              end
+
+              context "when the event is not processable" do
+                let(:status) { "HUBEE_NOTIFIED" }
+
+                it "does not update the shipment" do
+                  expect { interactor }.not_to change { shipment.reload.hubee_status }
+                end
+              end
+            end
+
+            context "when the event is not a status update" do
+              let(:action_type) { "ATTACH_DEPOSIT" }
+
+              it "deletes the notification" do
+                expect(session).to receive(:delete_notification)
+                interactor
+              end
+
+              it "does not update the event" do
+                expect(session).not_to receive(:update_event)
+                interactor
+              end
```

Comment:
J'fais un peu mon défenseur du DRY je sais, mais ici peut clairement factoriser ces deux it dans un `RSpec.shared_examples` pour pouvoir écrire juste `it_behaves_like "an ignored notification"`, vu qu'on s'en sert quand même 5 fois

Reply:
Done